pkg/scaffold/api.go

```diff
@@ -269,14 +277,24 @@ func (api *API) scaffoldV2() error {
 	return nil
 }
 
-// Since we support single group only in v2 scaffolding, validate if resource
-// being created belongs to existing group.
-func (api *API) validateResourceGroup(r *resource.Resource) error {
+// isNewGroup will return true if the group informed is diff of the pre-existents
+func (api *API) isNewGroup(r *resource.Resource) bool {
 	for _, existingGroup := range api.project.ResourceGroups() {
-		if !strings.EqualFold(r.Group, existingGroup) {
-			return fmt.Errorf("group '%s' is not same as existing group '%s'. Multiple groups are not supported yet.", r.Group, existingGroup)
+		if strings.ToLower(r.Group) != strings.ToLower(existingGroup) {
+			return true
 		}
 	}
+	return false
```

Comment:
Just to make it clear, it will work while project is single group as `api.project.ResourceGroups()` will just have one element but when a multigroup project is present we need to check every element not just the first.

Reply:
The func/check is just used/valid when multigroup is not enabled.
Added a comment in order to make it clear. So, has no issue with 

>  it will work while project is single group as api.project.ResourceGroups() will just have one element but when a multigroup project is present we need to check every element not just the first.

Because it is not used for the multigroup scenario at all.

Reply:
Just negate the check and swap `true` and `false` and it will work for both:

```golang
// isNewGroup will return true if the group informed is diff of the pre-existents
func (api *API) isNewGroup(r *resource.Resource) bool {
	for _, existingGroup := range api.project.ResourceGroups() {
		if strings.EqualFold(r.Group, existingGroup) {
			return false
		}
	}
	return true
}
```

Reply:
your suggestion will not work. Fixed for all scenarios any way.

Reply:
Why do you say it doens't work?

Imagine that we already have "group1", "group2" and "group3":
```
api.project.ResourceGroups() = []string{
    "group1",
    "group2",
    "group`3",
}
```

A new resource for `"group2"` is being created (`r.group = "group2"`):
1- `strings.EqualFold("group2", "group1")` is false, do not return anything
2- `strings.EqualFold("group2", "group2")` is true, return false
`isNewGroup() = false` as "group2" was already in the list of groups

A new resource for `"unknown"` is being created (`r.group = "unknown"`):
1- `strings.EqualFold("unknown", "group1")` is false, do not return anything
2- `strings.EqualFold("unknown", "group2")` is false, do not return anything
3- `strings.EqualFold("unknown", "group3")` is false, do not return anything
4- End of the loop, return true
`isNewGroup() = true` as "unknown" was not in the list of groups.